My project/Assets/Scripts/CreatureLogic/Actions/Wander.cs

```diff
@@ -23,20 +22,27 @@ public bool EndCondition()
         return false;
     }
 
+    /*
+     * On enter: 
+     * a random target coordinate is set
+     * the velocity is set such that it goes towards the coordinate
+     */
     public void OnEnter()
     {
         wander_target = data.SetRandomPath();
         while(wander_target == Vector3Int.zero){
             wander_target = data.SetRandomPath();
         }
-        wandering = true;
-        Debug.Log("Wander Target: " + wander_target);
+        Vector3Int grid_position = GameManager.Instance.getGrid().WorldToCell(rb.position);
+        rb.velocity = new Vector2(wander_target.x - grid_position.x, wander_target.y - grid_position.y).normalized * data.Speed * .02f;
```

Comment:
what is .02f? it might make sense to abstract this line into a reusable function with a name - it's not super clear to me what it's doing - especially with the .02f